tests/unit/query-cache-test.js

```diff
@@ -922,4 +922,52 @@ module('unit/query-cache', function(hooks) {
       assert.equal(this.adapterAjax.called, false, '`adapter.ajax` is not called');
     });
   });
+
+  test('.cacheURL inserts consistently inserts a promise into cache', function(assert) {
+    assert.expect(2);
+    const mockResult = { id: 1, foo: 'bar' };
+
+    this.queryCache.cacheURL('foo-bar', mockResult);
+    const cachedResult = this.queryCache.queryURL('/foo-bar', { cacheKey: 'foo-bar' });
+
+    assert.ok(cachedResult && cachedResult.then, 'cachedResult is a promise');
+
+    return cachedResult.then(result => {
+      assert.deepEqual(mockResult, result, 'queryURL returned cached result');
```

Comment:
can we add an assertion that `this.adapterAjax` was not called (similar to the `.queryURL caches its results when given a cacheKey` test)